packages/frontend/src/hooks/useTransactionReplacement.ts

```diff
@@ -0,0 +1,106 @@
+import { ContractTransaction } from '@ethersproject/contracts'
+import { errors } from 'ethers'
+import { useCallback, useEffect, useState } from 'react'
+import Transaction from 'src/models/Transaction'
+// import logger from 'src/logger'
+// import Network from 'src/models/Network'
+// import { Token } from '@hop-protocol/sdk'
+
+function useTransactionReplacement(
+  txHistory
+  // sdk
+  // fromNetwork?: Network,
+  // toNetwork?: Network,
+  // token?: Token
+) {
+  // const [transaction, setTransaction] = useState<ContractTransaction>()
+  // const [txModel, setTxModel] = useState<any>()
+  // const [watcher, setWatcher] = useState<any>()
+
+  const waitForTransaction = useCallback(
+    async (transaction: ContractTransaction, txModelArgs: any) => {
+      if (!transaction) return
+      // setTransaction(transaction)
+
+      try {
+        return await transaction.wait()
+      } catch (error: any) {
+        if (error.code === errors.TRANSACTION_REPLACED) {
+          const { replacement, receipt } = error
+          if (error.cancelled) {
+            // console.log(`error.cancelled__error.replacement:`, error.replacement)
+          } else {
+            console.log(`replaced transaction, receipt:`, replacement, receipt)
+            const replacementTxModel = new Transaction({
+              hash: error.replacement.hash,
+              replaced: true,
+              ...txModelArgs,
+            })
+
+            // Replace local storage
+            txHistory?.replaceTransaction(transaction.hash, replacementTxModel)
+
+            // if (watcher) {
+            //   watcher.off(sdk.Event.DestinationTxReceipt)
+            // }
+
+            // const { token, networkName, destNetworkName } = txModelArgs
+            // if (destNetworkName) {
+            //   const replacementWatcher = sdk.watch(
+            //     replacementTxModel.hash,
+            //     token.symbol,
+            //     networkName,
+            //     destNetworkName
+            //   )
+            //   setWatcher(replacementWatcher)
+            // }
+
+            return {
+              originalTx: transaction,
+              replacementTxModel,
+              replacementTx: replacement,
+              replacementReceipt: receipt,
+            }
+          }
+        }
+      }
+    },
+    [txHistory]
+  )
+
+  // useEffect(() => {
```

Comment:
this and other `watcher`-related code was an attempt to cleverly react to replaced txs. it was not a successful endeavor, and was commented out for documentation purposes. happy to remove if it's distracting.